### ISSUE TITLE
Interface for createContentScriptUi does not align with documentation

### ISSUE THREAD

**Issue by AndrewWalsh** [wxt-dev/wxt#267]
### Describe the bug

Very impressed with this framework btw, nice work.

This regards the method `createContentScriptUi` defined in https://wxt.dev/entrypoints/content-scripts.html#ui.

The docs refer to two methods `mount` and `onRemove`. There are a few discrepancies in the TypeScript interface:

- `mount` should take a parameter `container: Element`
- The interface defines a method `remove`, but not `onRemove`
- The docs say that `remove/onRemove` take a parameter `onRemove(app)`, but this isn't in the interface and I'm not sure what this should be

https://github.com/wxt-dev/wxt/blob/59094c9af3f01af271c0faf4e48fa69017e49df9/src/client/content-scripts/content-script-ui.ts#L118-L145

### Expected behavior

The interface aligns with the docs and these parameters are defined

**Comment by aklinker1** [comment]
Thanks, glad you like it! 😊

The docs for `createContentScriptUi` and `createContentScripIframe` could be cleaned up and improved... They are confusing, and a full example would help.

But it does behave as described.

I think the thing that's confusing you is that there's two mount functions. One gets passed into `createContentScriptUi`, the other is called from the return value.

The interface you shared is the type for the return type, which only includes `mount` and `remove`.

Right underneath it is the options interface that gets passed into the function.

https://github.com/wxt-dev/wxt/blob/59094c9af3f01af271c0faf4e48fa69017e49df9/src%2Fclient%2Fcontent-scripts%2Fcontent-script-ui.ts#L147-L173

This is the interface with `mount(container): app` and `onRemove(app)`.

> - `mount` should take a parameter `container: Element`

This mount function is passed into `createContentScriptUi`, and cannot be called from the return value. It is used to mount your UI inside the ShadowRoot.

Anchor and alignment are used to position your UI on the page. That is why the mount function on the return type provides no arguments - you've already defined where the UI will be mounted.

> - The interface defines a method `remove`, but not `onRemove`

onRemove is passed into ``
createContentScriptUi`, not called on the return value.

> - The docs say that `remove/onRemove` take a parameter `onRemove(app)`, but this isn't in the interface and I'm not sure what this should be

The `remove` function on the return value doesn't take any parameters because, once again, you already defined the onRemove behavior when calling the `creayeContentScriptUi`.

---

I will improve the docs and add an example. I'd really appreciate your feedback once complete, it's hard to judge how understandable docs you wrote are 😅

**Comment by AndrewWalsh** [comment]
@aklinker1 Thank you for the extremely detailed answer! You're right, I'm working in a monorepo and have made a mistake on my end here with my `tsconfig.json`. Please disregard this.

>I will improve the docs and add an example. I'd really appreciate your feedback once complete, it's hard to judge how understandable docs you wrote are 😅

Absolutely I'd be happy to take a look. I'll be using your lib quite a lot and will check back to see the update  👍 .

**Comment by aklinker1** [comment]
@AndrewWalsh I published the new docs. When you have time, let me know what you think!

https://wxt.dev/guide/content-script-ui.html

**Comment by AndrewWalsh** [comment]
@aklinker1 Saw your update, your comment makes sense and this definitely feels clearer. Few comments:

## Integrated

> Integrated content script UIs use the page's CSS to inject a UI that looks like it's apart of the page.
WXT doesn't provide any utils for mounting integrated UIs yet. Here are some examples for setting up integrated UIs.

I think the phrasing here `use the page's CSS to inject a UI that looks like it's apart of the page` could be clearer. Specifically, `use the page's CSS` is a bit ambiguous. So is `WXT doesn't provide any utils for mounting integrated UIs` as it appears to me that `defineContentScript` is doing just that. Perhaps instead this could be changed to something like:

> Integrated content scripts are injected alongside the content of a page. This means that they are affected by CSS on that page. You can control how CSS is injected for an integrated content script with the `cssInjectionMode` property.

The goal with this is to:
- Clarify that integrated scripts are affected by the styling on the page they are inject into, but not determined by it
- Clarify how integrated scripts can be styled

## ShadowRoot

- Under `ShadowRoot` there's text [`ShadowRoot` API](), looks like two separate links, perhaps better as [`ShadowRoot API`]()?
- Typo: effecting -> affecting
- `It isolates an element's style from the page's style.` suggestion -> `It isolates a content script's style from the page's style.`

---

I'm going to avoid using `createContentScriptUi` for now as I want to make use of HMR, but will get round to it 👍.

**Comment by aklinker1** [comment]
Great feedback! I'll update the docs today.

**Comment by aklinker1** [comment]
I've added your suggestions in https://github.com/wxt-dev/wxt/pull/272. 

I want to add a helper function for integrated UIs, similar to `createContentScriptUi` and `createContentScriptIframe`, but I'm recognizing that adding a third function would be confusing and that the API reference is hard to follow because of how I've defined the types.

So I think I'm going to simplify all that into a single function with a basic type for the options. Something like this:

```ts
export function createContentScriptUi(
  options: {
    mode?: "integrated" | "shadow-root" | "iframe",
    position?: "inline" | "overlay" | "modal",
    zIndex?: number;
    anchor?: string | Element;
    alignment?: "top-left" | "top-right" | "bottom-right" | "bottom-left";
    page?: string;
    css?: string;
    name?: string;
    mount?(container: Element): TApp;
    onRemove?(app: TApp): void;
  },
): ContentScriptUi;
```
